### ISSUE TITLE
Empty for/while blocks without braces result in broken code when using --compact=true

### ISSUE THREAD

**Issue by ntuckerxx** [babel/babel#2621]
A `for` or `while` block containing an empty statement without braces causes the generated code to be invalid because the semicolon is lost:

input:
```javascript
function hello() {
    for(;;);
}
```

output (syntax error):
```javascript
"use strict";function hello(){for(;;)}
```

This is the root cause of the inability to use socket.io-client in the latest release of React Native.  Bug: https://github.com/facebook/react-native/issues/3763

**Comment by sebmck** [comment]
This was introduced via #2608. I'm reverting that commit and releasing a new version. cc @amasad.

**Comment by sebmck** [comment]
Released 5.8.33
